File: code/python/largescale/Stationary_Point.py
stationary_w = []
dimension = 0


def stationary_point_train_basic(points):
	global stationary_w, dimension

	dimension = 2
	stationary_w = [0] * dimension

	X = 0
	XX = 0
	Y = 0
	XY = 0
	N = len(points)

	for i in range(0, N):
		X += points[i][0]
		XX += points[i][0] ** 2
		Y += points[i][1]
		XY += points[i][0] * points[i][1]

	stationary_w[1] = (XY / X - Y / N) / (XX / X - X / N)
	stationary_w[0] = (Y - X * stationary_w[1]) / N
	print("{0} * t + {1}".format(stationary_w[1], stationary_w[0]))


def stationary_point_predict_basic(t):
	global stationary_w, dimension

	return stationary_w[1] * t + stationary_w[0]


def write_stationary_points():
	points = []
	t = 0
	with open("/home/hm/Documents/stationary_points.txt", "r") as fin:
		for line in fin:
			line = line[: len(line) - 1]
			points.append((t, int(line)))
			t += 1
	stationary_point_train_basic(points[500:])
	with open("/home/hm/Documents/stationary_points_predict.txt", "w") as fout:
		for i in range(0, len(points)):
			fout.write("{0}\n".format(stationary_point_predict_basic(i)))


if __name__ == "__main__":
	write_stationary_points()

File: code/java/src/FunctionApproximation/Stationary_Point.java
package FunctionApproximation;

import FunctionLibrary.FileOperation;
import FunctionLibrary.UltraFile;

public class Stationary_Point {
    public static double[] stationary_w;
    public static int dimension = 0;

    public static void stationary_point_train_basic(double[][] points) {
        dimension = 2;
        stationary_w = new double[dimension];

        double X = 0;
        double XX = 0;
        double Y = 0;
        double XY = 0;
        int N = points.length;

        for (int i = 0; i < N; ++i) {
            X += points[i][0];
            XX += points[i][0] * points[i][0];
            Y += points[i][1];
            XY += points[i][0] * points[i][1];
        }

        stationary_w[1] = (XY / X - Y / N) / (XX / X - X / N);
        stationary_w[0] = (Y - X * stationary_w[1]) / N;
        System.out.println(String.format("%f * t + %f", stationary_w[1], stationary_w[0]));
    }

    public static double stationary_point_predict_basic(double t){
        return stationary_w[1] * t + stationary_w[0];
    }


}
